### DISCUSSION TITLE
How to add listerner for minimize/maximize app?

### DISCUSSION THREAD

**Discussion by hadisn** [airsdk/Adobe-Runtime-Support#2407] [General]
I have using this for detect if user switch app:
  ```
        NativeApplication.nativeApplication.addEventListener(Event.DEACTIVATE, onMinimize);
	NativeApplication.nativeApplication.addEventListener(Event.ACTIVATE, onMaximize);
```
but when interstitial ad is open application detect it like the app is minimized, is there another way to detect if app is not maximized or switched to another app...?

**Comment by marchbold** [comment]
You can use our Application extension to get access to the background / foreground events:

https://docs.airnativeextensions.com/docs/application/application-state-events

These events are more accurate to the system putting your application into the background rather than just the application activity being deactivated by another activity (such as an interstitial ad).

**Reply by hadisn** [reply]
@marchbold Thank you, I already use your Adverts extension, would it be too much extensions for one application? :)

**Reply by marchbold** [reply]
@hadisn Adding the Application ANE wouldn't be an issue at all. We have lots of extensions in most of our apps :)
